### PR TITLE
allow for Job class to have an __await__ method

### PR DESCRIPTION
**Pull request by Vizonex**
I thought it would be a smart idea if the Job class could be something that could be optionally awaited on simply by just calling await in cases were code such as `await job.wait()` is not feasible and where we know in advance that we don't need a timeout variable inplace, this would allow for something such as `asyncio.gather()` to be easily done on a group of jobs to being awaited on rather than having to write a bunch of coroutines beforehand. This would make the job class more task-like since asyncio tasks have this same feature where they can be awaited simply by calling `await task` for instance. I have also written some unittests as well to for this library that would best demonstrate these examples.

## Related issue number

I don't think this is related to any issues however I do believe that this idea would be helpful.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [] Documentation reflects the changes

### CODE REVIEW

File: tests/test_job.py [hunk 1]
```diff
@@ -314,3 +314,23 @@ async def coro() -> None:
...
+    result = await job
+    assert result == 1
+
+
+async def test_job_gathering(scheduler: Scheduler):
+    async def coro() -> int:
+        """Dummy function with result"""
+        return 1
+
+    jobs = [scheduler.spawn(coro()) for _ in range(5)]
```
Comment (Dreamsorcerer) [inline tests/test_job.py:334]:
```suggestion
    jobs = (scheduler.spawn(coro()) for _ in range(5))
```

**Comment by Dreamsorcerer** [comment]
Yeah, I think this seems reasonable.

**Comment by Dreamsorcerer** [comment]
Except for all the issues that arise... So, maybe not..

**Comment by asvetlov** [comment]
No, it doesn't work this way.
`asyncio.gather()` requires futures or coroutines, it means that `Job` should be derived from `asyncio.Future`.
But job and future have completely different API and serve different roles, let's keep everything as is.

**Pull request closed without merging**